Sources/SegmentFirebase/FirebaseDestination.swift

```diff
@@ -103,8 +103,13 @@ public class FirebaseDestination: DestinationPlugin {
     public func screen(event: ScreenEvent) -> ScreenEvent? {
         
         if let eventName = event.name {
+            var parameters: [String: Any] = [:]
+            if let properties = event.properties?.dictionaryValue {
+                parameters = returnMappedFirebaseParameters(properties)
+            }
+            parameters[FirebaseAnalytics.AnalyticsParameterScreenName] = eventName
             FirebaseAnalytics.Analytics.logEvent(FirebaseAnalytics.AnalyticsEventScreenView,
-                                                 parameters: [FirebaseAnalytics.AnalyticsParameterScreenName: eventName])
+                                                 parameters: parameters)
```

Comment:
Interesting I didn't realize that `screen` calls didn't include any other properties than eventName. Maybe for now lets just make our change to the `track` call above ^.

It seems there was a misunderstanding with the requirements. We need to add a segment plugin like `BrazeExternalIDPlugin` called `UTMPlugin`. This plugin will hold UTMs and then modify the context by adding campaign values. See here: BrazeExternalIDPlugin.

Then in this repo I *believe* we need the ability to take values from `context` and pass them through to firebase every time `track` is called because right now nothing from `context` is passed through